SimpleGame/src/main/scala-2.11/edu/gemini/experimental/jluhrs/ConsoleKeys.scala

```diff
@@ -17,25 +19,34 @@ object ConsoleKeys {
 
   val inputP: Process[Task, Event[PlayerInput]] = inputQ.dequeue
 
+  private val continue = new AtomicBoolean(true)
+
   private val keySource: Task[Unit] = Task {
 
     // Some munbo jumbo to get keystrokes unbuffered.
     (Seq("sh", "-c", "stty -icanon min 1 < /dev/tty") !)
     (Seq("sh", "-c", "stty -echo < /dev/tty") !)
 
-    while (true) {
-      val c = Console.in.read
-      c match {
-        case 119 => inputQ.enqueueOne((System.nanoTime nanoseconds, Up)).unsafePerformSync
-        case  97 => inputQ.enqueueOne((System.nanoTime nanoseconds, Left)).unsafePerformSync
-        case 115 => inputQ.enqueueOne((System.nanoTime nanoseconds, Down)).unsafePerformSync
-        case 100 => inputQ.enqueueOne((System.nanoTime nanoseconds, Right)).unsafePerformSync
-        case _ => ()
+    while (continue.get) {
```

Comment:
This will likely work but I wonder if there is some other way, sort of sending a poison pill to the stream and it would terminate itself